src/core/meshcop/dtls.cpp

```diff
@@ -295,6 +295,8 @@ otError Dtls::Setup(bool aClient)
     mbedtls_ssl_conf_rng(&mConf, mbedtls_ctr_drbg_random, Random::Crypto::MbedTlsContextGet());
     mbedtls_ssl_conf_min_version(&mConf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
     mbedtls_ssl_conf_max_version(&mConf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
+
+    assert(mCipherSuites[1] == 0);
```

Comment:
do we need assert here?

Reply:
Add `assert` here to ensure we always remember to set this,

Prefer `assert` because `mCipherSuites` is not exposed and won't be changed by user inputs. So a assertion failure indicates a programming bug.